src/olympia/versions/models.py

```diff
@@ -283,7 +282,11 @@ def from_upload(
         if upload.addon and upload.addon != addon:
             raise VersionCreateError('FileUpload was made for a different Addon')
 
-        if not upload.user or not upload.ip_address or not upload.source:
+        if (
+            not getattr(upload, 'user', None)
```

Comment:
It always does, but when `upload.user` is `None`, django raises an exception (because I'm no longer allowing that in the model). The `getattr` is just to avoid this so I can raise my own (it's really a use case that shouldn't be happening any more, but I left this condition to be a bit paranoid)